src/main/java/org/terracotta/statistics/derived/latency/DefaultLatencyHistogramStatistic.java

```diff
@@ -0,0 +1,187 @@
+        return Double.isNaN(v) ? null : new long[]{(long) v, (long) v};
+      }
+      double[] bounds = histogram.getQuantileBounds(percent);
+      if (Double.isNaN(bounds[0]) || Double.isNaN(bounds[1])) {
+        return null;
+      }
+      return new long[]{(long) bounds[0], (long) nextDown(bounds[1])};
+    }
+
+    @Override
+    public List<Histogram.Bucket> buckets() {
+      return histogram.getBuckets();
+    }
+  };
+
+  private long nextPruning;
+
+  /**
+   * Create a histogram maintained over a sliding time window.
+   * <p>
+   * The constructed histogram is:
+   * </p>
+   * <ul>
+   * <li>maintained over {@code window} sliding window</li>
+   * <li>consists of {@code bucketCount} buckets</li>
+   * <li>where {@code b1.size() ~= b0.size * phi}</li>
+   * </ul>
+   * If "phi" is high, the quantile bounds will be more precise for lower percentiles such as minimum.
+   * If "phi" is low, the quantile bounds will be more precise for higher percentiles such as 99%-ile, maximum.
+   *
+   * @param phi         histogram bucket bias factor
+   * @param bucketCount number of buckets
+   * @param window      sliding window size, in ns
+   * @param timeSupplier      the supplier of time, which must be in the same unit as the time passed to the {{@link #event(long, long)}} method.
+   */
+  public DefaultLatencyHistogramStatistic(
+      double phi,
+      int bucketCount,
+      Duration window,
+      LongSupplier timeSupplier) {
+    this.timeSupplier = timeSupplier;
+    this.histogram = new BarSplittingBiasedHistogram(phi, bucketCount, window.toNanos());
+    this.pruningDelay = window.toNanos() / 2;
+  }
+
+  public DefaultLatencyHistogramStatistic(double phi,
+                                          int bucketCount,
+                                          Duration window) {
+    this(phi, bucketCount, window, Time::time);
+  }
+
+  @Override
+  public synchronized List<org.terracotta.statistics.derived.histogram.Histogram.Bucket> buckets() {
+    tryExpire(true, timeSupplier);
+    return multiQuery.buckets();
+  }
+
+  @Override
+  public long count() {
+    return multiQuery.count();
+  }
+
+  @Override
+  public synchronized Long minimum() {
+    tryExpire(true, timeSupplier);
+    return multiQuery.minimum();
+  }
+
+  @Override
+  public synchronized Long maximum() {
+    tryExpire(true, timeSupplier);
+    return multiQuery.maximum();
+
+  }
+
+  @Override
+  public synchronized Long percentile(double percent) {
+    tryExpire(true, timeSupplier);
+    return multiQuery.percentile(percent);
+  }
+
+  @Override
+  public synchronized long[] percentileBounds(double percent) {
+    tryExpire(true, timeSupplier);
+    return multiQuery.percentileBounds(percent);
+  }
+
+  @Override
+  public synchronized void event(long time, long latency) {
+    histogram.event(latency, time);
+    tryExpire(false, () -> time);
+  }
+
+  @Override
+  public synchronized <T> T multiQuery(Function<Query, T> fn) {
```

Comment:
This isn't a multiQuery, it's just a query - the multi query is tied to the type of T.